src/material-experimental/mdc-input/harness/input-harness.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
+import {InputHarnessFilters} from './input-harness-filters';
+
+/**
+ * Harness for interacting with a standard Material inputs in tests.
+ * @dynamic
+ */
+export class MatInputHarness extends ComponentHarness {
+  static hostSelector = '[matInput]';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for an input with
+   * specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `name` finds an input with specific name.
+   *   - `id` finds an input with specific id.
+   *   - `value` finds an input with specific value.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: InputHarnessFilters = {}): HarnessPredicate<MatInputHarness> {
+    return new HarnessPredicate(MatInputHarness)
+        .addOption(
```

Comment:
add a note that `name` and `id` can be removed when https://github.com/angular/components/pull/16848 goes in

Reply:
Done.